presto-main/src/test/java/com/facebook/presto/operator/TestTableFinishOperator.java

```diff
@@ -153,10 +165,112 @@ public void testStatisticsAggregation()
         assertThat(tableFinishInfo.getStatisticsWallTime().getValue(NANOSECONDS)).isGreaterThan(0);
         assertThat(tableFinishInfo.getStatisticsCpuTime().getValue(NANOSECONDS)).isGreaterThan(0);
 
+        assertTrue(lifespanCommitter.getPartitionIds().isEmpty());
+        assertTrue(lifespanCommitter.getFragmentsList().isEmpty());
+
+        assertEquals(driverContext.getSystemMemoryUsage(), 0);
+        assertEquals(driverContext.getMemoryUsage(), 0);
+    }
+
+    @Test
+    public void testRecoverability()
+            throws Exception
+    {
+        TestTableFinisher tableFinisher = new TestTableFinisher();
+        TestPartitionCommitter lifespanCommitter = new TestPartitionCommitter();
+        ColumnStatisticMetadata statisticMetadata = new ColumnStatisticMetadata("column", MAX_VALUE);
+        StatisticAggregationsDescriptor<Integer> descriptor = new StatisticAggregationsDescriptor<>(
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(statisticMetadata, 0));
+        Session session = testSessionBuilder()
+                .setSystemProperty("statistics_cpu_timer_enabled", "true")
+                .build();
+        TableFinishOperatorFactory operatorFactory = new TableFinishOperatorFactory(
+                0,
+                new PlanNodeId("node"),
+                tableFinisher,
+                lifespanCommitter,
+                new AggregationOperator.AggregationOperatorFactory(
+                        1,
+                        new PlanNodeId("test"),
+                        AggregationNode.Step.SINGLE,
+                        ImmutableList.of(LONG_MAX.bind(ImmutableList.of(STATS_START_CHANNEL), Optional.empty())),
+                        true),
+                descriptor,
+                session,
+                TABLE_COMMIT_CONTEXT_CODEC);
+        DriverContext driverContext = createTaskContext(scheduledExecutor, scheduledExecutor, session)
+                .addPipelineContext(0, true, true, false)
+                .addDriverContext();
+        TableFinishOperator operator = (TableFinishOperator) operatorFactory.createOperator(driverContext);
+
+        List<Type> inputTypes = ImmutableList.of(BIGINT, VARBINARY, VARBINARY, BIGINT);
+
+        // Non-grouped execution
```

Comment:
// pages for non-grouped execution
// expect lifespan committer will not be called